### DISCUSSION TITLE
Daily Shadow Pattern not responding?

### DISCUSSION THREAD

**Discussion by AnneSawa** [UMEP-dev/UMEP#285] [Q&A] [answered]
Hi, 

I have a dsm raster that I would like to calculate the Daily Shadow Pattern for. When I click run, the 
When I try to use the Daily Shadow Pattern tool, QGIS stops responding. It uses almost all of my working memory and I am not sure how to fix it. 

I am using QGIS 3.10.14.

**Comment by biglimp** [comment, marked as answer]
I suspect that this is not an issue so I will move this over too discussions as others could benefit from our discussion.

Three things:
1. What version of UMEP and what operating system are you using?
2. How large is your DSM (rows and columns)? It sounds that you are using a too big raster for your computer memory
3. Try using **UMEP for processing** where the shadow tool also is available. Here, QGIS automatically make use of a so called ”thread” which hi ders your computer to freeze.

Let me know your progress.
/Fredrik

**Reply by AnneSawa** [reply]
Hi, 

Thanks for your quick response!

1. I am using UMEP 3.17 and using Windows 10 64 bit 
2. DSM has 25509 (width) and 23386 (height)
3. I tried using this one and it still froze. 

edit: Also just realized the difference between _issue_ and _discussion_ (this was my first GitHub post)

**Reply by biglimp** [reply]
Your DSM is way too big for this application. I would not go above 4000*4000 pixels. The shadow casting algorithm used has a computational complexity of O^n where n i number of pixels. This means that a double sized DSM will take exponentionally longer to calculate. If you want to examine that large area you need to tile your DSM into smaller grids and then mosaic them back together afterwards. Remember to have a small overlap since shadow can travel based on Sun position and height of objects.
